### PR TITLE
Perform tab check later and safer

### PR DESCRIPTION
**Pull request by SlayterDev**
**Reviewer:** @GioSensation @sammacbeth 

## Description:

`thisTab` was being checked in cases where it didn't exist. This PR moves it a bit later and performs an extra check for the tab.

## Steps to test this PR:
1. Go to https://www.wga.com/users/ron-ratto
2. Click on the Pinnacle logo
3. The page opens in a new tab
4. Open the popup → 2 trackers found

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/background/redirect.es6.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ function handleRequest (requestData) {
 
     let thisTab = tabManager.get(requestData)
 
-    const blockingEnabled = utils.isFeatureEnabled('contentBlocking') && !utils.isFeatureBrokenForURL(thisTab.url, 'contentBlocking')
+    let blockingEnabled = utils.isFeatureEnabled('contentBlocking')
```
Comment (sammacbeth) [inline shared/js/background/redirect.es6.js:51]: We don't use this variable until the `else` that starts on line 77. We can just move the whole line down there after the `thisTab` check that is done at the start of that block.

**Review by sammacbeth — CHANGES_REQUESTED** [review]

File: shared/js/background/redirect.es6.js [hunk 2]
```diff
@@ -57,6 +57,10 @@ function handleRequest (requestData) {
         }
     }
 
+    if (thisTab) {
+        blockingEnabled = !utils.isFeatureBrokenForURL(thisTab.url, 'contentBlocking')
```
Comment (jonathanKingston) [inline shared/js/background/redirect.es6.js:61]: This should be an `&&` as before.

File: shared/js/background/redirect.es6.js [hunk 3]
```diff
@@ -48,7 +48,7 @@ function handleRequest (requestData) {
 
     let thisTab = tabManager.get(requestData)
```
Comment (jonathanKingston) [inline shared/js/background/redirect.es6.js:49]: Perhaps we should actually address why _blank is triggering the tab not being loaded into the tabManager as this seems a bigger concern to me.

Comment (sammacbeth) [reply shared/js/background/redirect.es6.js:49]: I think the error happens on the `main_frame` request, which further down should create the tab object.

**Review by sammacbeth — APPROVED** [review]

File: shared/js/background/redirect.es6.js (hunk 3 above)
Comment (jonathanKingston) [reply shared/js/background/redirect.es6.js:49]: Yeah just seen that actually thanks :)

**Pull request merged**